src/site/content/en/blog/generic-sensor/index.md

```diff
@@ -0,0 +1,478 @@
+  </figure>
+</div>
+
+The [`AbsoluteOrientationSensor`](https://w3c.github.io/orientation-sensor/#orientationsensor-interface)
+is a fusion sensor that measures rotation of a device in relation to the
+Earth's coordinate system, while the
+[`RelativeOrientationSensor`](https://w3c.github.io/orientation-sensor/#relativeorientationsensor-interface)
+provides data representing rotation of a device hosting motion sensors in
+relation to a stationary reference coordinate system.
+
+All modern 3D JavaScript frameworks support [quaternions](https://en.wikipedia.org/wiki/Quaternion)
+and [rotation matrices](https://en.wikipedia.org/wiki/Rotation_matrix)
+to represent rotation; however, if you use WebGL directly,
+the [`OrientationSensor`](https://w3c.github.io/orientation-sensor/#orientationsensor-populatematrix)
+interface has convenient methods for WebGL compatible rotation matrices.
+Here are few snippets:
+
+**[three.js](https://threejs.org/docs/index.html#api/core/Object3D.quaternion)**
+
+```js
+let torusGeometry = new THREE.TorusGeometry(7, 1.6, 4, 3, 6.3);
+let material = new THREE.MeshBasicMaterial({ color: 0x0071C5 });
+let torus = new THREE.Mesh(torusGeometry, material);
+scene.add(torus);
+
+// Update mesh rotation using quaternion.
+const sensorAbs = new AbsoluteOrientationSensor();
+sensorAbs.onreading = () => torus.quaternion.fromArray(sensorAbs.quaternion);
+sensorAbs.start();
+
+// Update mesh rotation using rotation matrix.
+const sensorRel = new RelativeOrientationSensor();
+let rotationMatrix = new Float32Array(16);
+sensor_rel.onreading = () => {
+    sensorRel.populateMatrix(rotationMatrix);
+    torus.matrix.fromArray(rotationMatrix);
+}
+sensorRel.start();
+```
+
+**[BABYLON](http://doc.babylonjs.com/classes/3.0/abstractmesh#rotationquaternion-quaternion-classes-3-0-quaternion-)**
+
+```js
+const mesh = new BABYLON.Mesh.CreateCylinder("mesh", 0.9, 0.3, 0.6, 9, 1, scene);
+const sensorRel = new RelativeOrientationSensor({frequency: 30});
+sensorRel.onreading = () => mesh.rotationQuaternion.FromArray(sensorRel.quaternion);
+sensorRel.start();
+```
+
+**[WebGL](https://www.khronos.org/registry/webgl/specs/latest/2.0/)**
+
+```js
+// Initialize sensor and update model matrix when new reading is available.
+let modMatrix = new Float32Array([1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1]);
+const sensorAbs = new AbsoluteOrientationSensor({frequency: 60});
+sensorAbs.onreading = () => sensorAbs.populateMatrix(modMatrix);
+sensorAbs.start();
+
+// Somewhere in rendering code, update vertex shader attribute for the model
+gl.uniformMatrix4fv(modMatrixAttr, false, modMatrix);
+```
+
+Orientation sensors enable various use cases, such as immersive gaming,
+augmented and virtual reality.
+
+For more information about motion sensors, advanced use cases, and requirements,
+please check the [motion sensors explainer](https://www.w3.org/TR/motion-sensors/)
```

Comment:
When this article was originally written, this was the only option. Four years later, is there a better link?

Reply:
We can point to the slightly more recent editor's draft.